packages/common/src/types/pivot.ts

```diff
@@ -6,37 +6,58 @@ export type PivotConfig = {
     metricsAsRows: boolean;
     columnOrder?: string[];
     hiddenMetricFieldIds?: string[];
+
+    columnTotals?: boolean;
+    rowTotals?: boolean;
 };
 
-type HeaderOrIndexType =
+type PTField =
     | { type: FieldType.METRIC; fieldId?: undefined }
     | { type: FieldType.DIMENSION; fieldId: string };
 
-export type PivotHeaderType = HeaderOrIndexType;
-export type PivotIndexType = HeaderOrIndexType;
+export type PTFieldLabel = {
+    type: 'label';
+    fieldId: string;
+    value?: undefined;
+};
+
+export type PTTotalLabel = {
+    type: 'total';
+};
 
-export type PivotValue =
-    | { type: 'label'; fieldId: string; value?: undefined }
-    | { type: 'value'; fieldId: string; value: ResultValue };
+export type PTValue = {
+    type: 'value';
+    fieldId: string;
+    value: ResultValue;
+};
 
-export type PivotTitleValue = PivotValue & {
+export type PTLabel = {
     titleDirection: 'index' | 'header';
 };
 
-export type PivotData = {
-    titleFields: (PivotTitleValue | null)[][];
+export type PTTotalValue = Pick<ResultValue, 'raw'>;
 
-    headerValueTypes: PivotHeaderType[];
-    headerValues: PivotValue[][];
+export type PTTitleValue = PTFieldLabel & PTLabel;
 
-    indexValueTypes: PivotIndexType[];
-    indexValues: PivotValue[][];
+export type PTTotalOrFieldLabel = PTLabel & (PTTotalLabel | PTFieldLabel);
+
+export type PivotData = {
+    headerValueTypes: PTField[];
+    headerValues: (PTFieldLabel | PTValue)[][];
+
+    indexValueTypes: PTField[];
+    indexValues: (PTFieldLabel | PTValue)[][];
 
     dataColumnCount: number;
     dataValues: (ResultValue | null)[][];
 
-    columnTotals?: (ResultValue | null)[][];
-    rowTotals?: (ResultValue | null)[][];
+    titleFields: (PTTitleValue | null)[][];
+
+    rowTotalHeaders?: (PTTotalOrFieldLabel | null)[][];
```

Comment:
Not sure why `titleDirection` is needed in these cases since we know if it's a row or column total

Reply:
yep, removing